src/NServiceBus.Core/Receiving/MainOnCompleteExecutor.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus.Transport;
+
+    class MainOnCompleteExecutor
```

Comment:
I added "Main" to follow the convention of https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs since this only triggers on the main receiver and not on the satellite receivers?

Reply:
Lets discuss tomorrow